AdyenCard/Utilities/BinLookup/BinLookupService.swift

```diff
@@ -40,6 +42,18 @@ internal final class BinLookupService: AnyBinLookupService {
         }
         
         let request = BinLookupRequest(encryptedBin: encryptedBin, supportedBrands: supportedCardTypes)
-        apiClient.perform(request, completionHandler: caller)
+        if let cached = cache[bin] {
```

Comment:
can we check for a cached `bin` in line 37 before trying to encrypt it and creating a  `BinLookupRequest`?

Reply:
Yes, we actually can!
Good one :)